src/trunk.jl

```diff
@@ -17,19 +22,62 @@ The nonmonotone strategy follows Section 10.1.3, Algorithm 10.1.2.
     SIAM, Philadelphia, USA, 2000.
     DOI: 10.1137/1.9780898719857.
 """
-function trunk(
+mutable struct TrunkSolver{T, V <: AbstractVector{T}, Op <: AbstractLinearOperator{T}}
+  x::V
+  xt::V
+  gx::V
+  gt::V
+  gn::V
+  Hs::V
+  subsolver::CgSolver{T, V}
```

Comment:
Could this be another solver?

Reply:
Potentially. It looks like I would have to use the specific solver call for each solver. `cg!` for CgSolver`, etc. @amontoison may correct me.

Reply:
You could use a `KrylovSolver{T, V}` for the subsolver. (https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/main/src/krylov_solvers.jl#L8)

Reply:
But how do I call the solver?

Reply:
If you need a generic function after to call the Krylov method, you could use that : 
```julia
const SOLVER = Dict(
  LsmrSolver           => :lsmr!
  CgsSolver            => :cgs!
  UsymlqSolver         => :usymlq!
  LnlqSolver           => :lnlq!
  BicgstabSolver       => :bicgstab!
  CrlsSolver           => :crls!
  LsqrSolver           => :lsqr!
  MinresSolver         => :minres!
  CgneSolver           => :cgne!
  DqgmresSolver        => :dqgmres!
  SymmlqSolver         => :symmlq!
  TrimrSolver          => :trimr!
  UsymqrSolver         => :usymqr!
  BilqrSolver          => :bilqr!
  CrSolver             => :cr!
  CraigmrSolver        => :craigmr!
  TricgSolver          => :tricg!
  CraigSolver          => :craig!
  DiomSolver           => :diom!
  LslqSolver           => :lslq!
  TrilqrSolver         => :trilqr!
  CrmrSolver           => :crmr!
  CgSolver             => :cg!
  CgLanczosShiftSolver => :cg_lanczos!
  CglsSolver           => :cgls!
  CgLanczosSolver      => :cg_lanczos!
  BilqSolver           => :bilq!
  MinresQlpSolver      => :minres_qlp!
  QmrSolver            => :qmr!
)

function krylov!(solver :: KS, args...; kwargs...) where KS <: KrylovSolver
  fn = SOLVER[KS.name.wrapper]
  @eval $fn(solver, $args...; $kwargs...)
end
```

Reply:
This could probably go on Krylov. I would recommend something like `solve!` instead of `krylov!` to generalize for KrylovSolvers and JSOSolvers. @dpo, how about we generalize in a future PR?

Reply:
We had a discussion about it with @tmigot last month and I forgot to open a pull request.

Reply:
Dispatch would be better here